generators/entity/index.js

```diff
@@ -601,6 +601,10 @@ class EntityGenerator extends BaseBlueprintGenerator {
         this.context.entityContainsCollectionField = this.context.relationships.some(relationship => relationship.relationshipCollection);
       },
 
+      processPrimaryKeyTypesForRelations() {
+        this.context.otherEntityPrimaryKeyTypes = new Set(this.context.relationships.map(relationship => relationship.otherEntity.primaryKey.type));
```

Comment:
```suggestion
        this.context.otherEntityPrimaryKeyTypes = new Set(this.context.relationships.map(relationship => relationship.otherEntity.primaryKey && relationship.otherEntity.primaryKey.type));
```
- primaryKey doesn't exists for embedded entities.
- IMO you should add Set to `otherEntityPrimaryKeyTypes` or convert to back to array.
Every field is array, so it can be confusing to use has instead of includes.